# Filtering Dataset web use

1. Filter Request URI mengambil HTMLnya saja
2. Filter Status mengambil nilai yg 200
3. Filter Request Method mengambil yg GET
4. Merubah Request Time menjadi tanggal dan waktunya saja
5. Filter Berdasarkan IP Address(Remote host) dan User Agent(Request Protocol)

## load dataset

In [30]:
import pandas as pd

# Upload file
from google.colab import files
uploaded = files.upload()

# Membaca file CSV hasil upload
df = pd.read_csv(next(iter(uploaded)), encoding='utf-8')

print("Data berhasil dimuat:", df.shape)
df.head()

Saving webuage.csv to webuage.csv
Data berhasil dimuat: (132258, 9)


,Remote host,Remote logname,Remote user,Request time,Request method,Request URI,Request Protocol,Status,Size of response (incl. headers)
0,65.55.147.227,-,-,2009-10-15T02:00:24Z,GET,/index.html,HTTP/1.1,200,21878
1,65.55.86.34,-,-,2009-10-15T02:00:58Z,GET,/index.html,HTTP/1.1,200,1416
2,148.188.55.88,-,-,2009-10-15T02:01:41Z,GET,/faq.html,HTTP/1.1,200,10946
3,72.30.57.238,-,-,2009-10-15T02:01:59Z,GET,/contribute.txt,HTTP/1.0,200,39943
4,66.249.139.233,-,-,2009-10-15T02:02:09Z,GET,/faq.html,HTTP/1.1,200,17247


In [31]:
df.columns

Index(['Remote host', 'Remote logname', 'Remote user', 'Request time',
       'Request method', 'Request URI', 'Request Protocol', 'Status',
       'Size of response (incl. headers)'],
      dtype='object')

## filter request URI (hanya html)

In [33]:
print("Jumlah data sebelum filter:", len(df))

df_html = df[df['Request URI'].str.endswith('.html', na=False)]

print("Jumlah data sesudah filter:", len(df_html))

df_html.head()

Jumlah data sebelum filter: 132258
Jumlah data sesudah filter: 79176


,Remote host,Remote logname,Remote user,Request time,Request method,Request URI,Request Protocol,Status,Size of response (incl. headers)
0,65.55.147.227,-,-,2009-10-15T02:00:24Z,GET,/index.html,HTTP/1.1,200,21878
1,65.55.86.34,-,-,2009-10-15T02:00:58Z,GET,/index.html,HTTP/1.1,200,1416
2,148.188.55.88,-,-,2009-10-15T02:01:41Z,GET,/faq.html,HTTP/1.1,200,10946
4,66.249.139.233,-,-,2009-10-15T02:02:09Z,GET,/faq.html,HTTP/1.1,200,17247
5,72.30.50.248,-,-,2009-10-15T02:02:13Z,GET,/index.html,HTTP/1.0,200,7883


## filter mengambil nilai yang 200

In [35]:
print("Jumlah data sebelum filter:", len(df))

# Filter data dengan status response == 200
df_status_200 = df[df['Status'] == 200]

print("Jumlah data sesudah filter:", len(df_status200))
df_status200.head()

Jumlah data sebelum filter: 132258
Jumlah data sesudah filter: 951345


,Unnamed: 0,host,time,method,url,response,bytes
0,0,***.novo.dk,805465029,GET,/ksc.html,200,7067
1,1,***.novo.dk,805465031,GET,/images/ksclogo-medium.gif,200,5866
2,2,***.novo.dk,805465051,GET,/images/MOSAIC-logosmall.gif,200,363
3,3,***.novo.dk,805465053,GET,/images/USA-logosmall.gif,200,234
4,4,***.novo.dk,805465054,GET,/images/NASA-logosmall.gif,200,786


## filter mengambil GET

In [36]:
print("Jumlah data sebelum filter:", len(df))

# Filter baris yang method-nya GET
df_get = df[df['Request method'] == 'GET']

print("Jumlah data sesudah filter:", len(df_get))
df_get.head()

Jumlah data sebelum filter: 132258
Jumlah data sesudah filter: 132147


,Remote host,Remote logname,Remote user,Request time,Request method,Request URI,Request Protocol,Status,Size of response (incl. headers)
0,65.55.147.227,-,-,2009-10-15T02:00:24Z,GET,/index.html,HTTP/1.1,200,21878
1,65.55.86.34,-,-,2009-10-15T02:00:58Z,GET,/index.html,HTTP/1.1,200,1416
2,148.188.55.88,-,-,2009-10-15T02:01:41Z,GET,/faq.html,HTTP/1.1,200,10946
3,72.30.57.238,-,-,2009-10-15T02:01:59Z,GET,/contribute.txt,HTTP/1.0,200,39943
4,66.249.139.233,-,-,2009-10-15T02:02:09Z,GET,/faq.html,HTTP/1.1,200,17247


## merubah request time menjadi tanggal dan waktu

In [38]:
import pandas as pd

print("Jumlah data sebelum transformasi waktu:", len(df))

# Konversi kolom time menjadi datetime
df['Request time'] = pd.to_datetime(df['Request time'], errors='coerce')

# Pisahkan menjadi tanggal dan waktu saja
df['tanggal'] = df['Request time'].dt.date
df['waktu'] = df['Request time'].dt.time

print("Jumlah data sesudah transformasi (baris sama):", len(df))

df[['Request time']].head(10)

Jumlah data sebelum transformasi waktu: 132258
Jumlah data sesudah transformasi (baris sama): 132258


,Request time
0,2009-10-15 02:00:24+00:00
1,2009-10-15 02:00:58+00:00
2,2009-10-15 02:01:41+00:00
3,2009-10-15 02:01:59+00:00
4,2009-10-15 02:02:09+00:00
5,2009-10-15 02:02:13+00:00
6,2009-10-15 02:02:37+00:00
7,2009-10-15 02:02:39+00:00
8,2009-10-15 02:02:51+00:00
9,2009-10-15 02:02:54+00:00


# Filtering Dataset NASA

1. Filter Request URI mengambil HTMLnya saja
2. Filter Status mengambil nilai yg 200
3. Filter Request Method mengambil yg GET
4. Merubah Request Time menjadi tanggal dan waktunya saja
5. Filter Berdasarkan IP Address(Remote host) dan User Agent(Request Protocol)

In [20]:
import pandas as pd

# Upload file
from google.colab import files
uploaded = files.upload()

# Membaca file CSV hasil upload
df = pd.read_csv(
    next(iter(uploaded)),
    sep=None,
    engine='python',
    on_bad_lines='skip'
)


print("Data berhasil dimuat:", df.shape)
df.head()

Saving data.csv to data (3).csv
Data berhasil dimuat: (1048575, 7)


,Unnamed: 0,host,time,method,url,response,bytes
0,0,***.novo.dk,805465029,GET,/ksc.html,200,7067
1,1,***.novo.dk,805465031,GET,/images/ksclogo-medium.gif,200,5866
2,2,***.novo.dk,805465051,GET,/images/MOSAIC-logosmall.gif,200,363
3,3,***.novo.dk,805465053,GET,/images/USA-logosmall.gif,200,234
4,4,***.novo.dk,805465054,GET,/images/NASA-logosmall.gif,200,786


In [23]:
df.columns

Index(['Unnamed: 0', 'host', 'time', 'method', 'url', 'response', 'bytes'], dtype='object')

## FILTER REQUEST URI YANG .html SAJA

In [25]:
print("Jumlah data sebelum filter:", len(df))

# Filter URL yang berakhiran .html
df_html = df[df['url'].str.endswith('.html', na=False)]

print("Jumlah data sesudah filter:", len(df_html))
df_html.head()

Jumlah data sebelum filter: 1048575
Jumlah data sesudah filter: 220905


,Unnamed: 0,host,time,method,url,response,bytes
0,0,***.novo.dk,805465029,GET,/ksc.html,200,7067
6,6,***.novo.dk,805465068,GET,/shuttle/missions/missions.html,200,8678
12,12,***.novo.dk,805465381,GET,/shuttle/resources/orbiters/columbia.html,200,6922
13,13,***.novo.dk,807951768,GET,/shuttle/missions/sts-69/mission-sts-69.html,200,11264
23,23,***.novo.dk,807951938,GET,/shuttle/countdown/liftoff.html,200,4665


Langkah kode tersebut memiliki fungsi sebagai berikut:

- len(df) menghitung jumlah baris keseluruhan sebelum proses filter sehingga kita tahu ukuran data mentah.

- df['url'].str.endswith('.html', na=False) merupakan proses inti filtering yang mengecek apakah sebuah URL benar-benar berakhiran .html. Jika iya, baris tersebut dipertahankan.

- na=False memastikan bahwa URL yang kosong atau nilai NaN tidak menyebabkan error dan otomatis dianggap tidak sesuai filter.

- Hasil filtering disimpan pada df_html, lalu kode menampilkan jumlah baris setelah filter dan beberapa contoh baris pertama menggunakan head().

Dari output yang dihasilkan:

- Jumlah data sebelum filter: 1,048,575 baris

- Jumlah data sesudah filter (hanya URL .html): 220,905 baris

- Hasil ini menunjukkan bahwa hanya sekitar 21% dari seluruh request yang benar-benar mengarah ke halaman HTML. Baris-baris yang ditampilkan pada df_html.head() memperlihatkan contoh URL seperti /ksc.html, /shuttle/missions/missions.html, dan lain-lain, yang berarti filter berjalan sesuai harapan.


## FILTER STATUS CODE 200

In [26]:
print("Jumlah data sebelum filter:", len(df))

# Filter data dengan status response == 200
df_status200 = df[df['response'] == 200]

print("Jumlah data sesudah filter:", len(df_status200))
df_status200.head()

Jumlah data sebelum filter: 1048575
Jumlah data sesudah filter: 951345


,Unnamed: 0,host,time,method,url,response,bytes
0,0,***.novo.dk,805465029,GET,/ksc.html,200,7067
1,1,***.novo.dk,805465031,GET,/images/ksclogo-medium.gif,200,5866
2,2,***.novo.dk,805465051,GET,/images/MOSAIC-logosmall.gif,200,363
3,3,***.novo.dk,805465053,GET,/images/USA-logosmall.gif,200,234
4,4,***.novo.dk,805465054,GET,/images/NASA-logosmall.gif,200,786


Dari output yang dihasilkan:

- Jumlah data sebelum filter: 1,048,575 baris

- Jumlah data sesudah filter (status 200): 951,345 baris

Artinya, sekitar 90,8% request berhasil diproses dengan baik oleh server. Ini menunjukkan bahwa sebagian besar permintaan yang diterima server tidak mengalami error seperti 404 (Not Found) atau 500 (Server Error).

## FILTER REQUEST METHOD GET

In [27]:
print("Jumlah data sebelum filter:", len(df))

# Filter baris yang method-nya GET
df_get = df[df['method'] == 'GET']

print("Jumlah data sesudah filter:", len(df_get))
df_get.head()

Jumlah data sebelum filter: 1048575
Jumlah data sesudah filter: 1043273


,Unnamed: 0,host,time,method,url,response,bytes
0,0,***.novo.dk,805465029,GET,/ksc.html,200,7067
1,1,***.novo.dk,805465031,GET,/images/ksclogo-medium.gif,200,5866
2,2,***.novo.dk,805465051,GET,/images/MOSAIC-logosmall.gif,200,363
3,3,***.novo.dk,805465053,GET,/images/USA-logosmall.gif,200,234
4,4,***.novo.dk,805465054,GET,/images/NASA-logosmall.gif,200,786


Penjelasan bagian kode:

- len(df) menampilkan jumlah data sebelum difilter.

- df[df['method'] == "GET"] mengambil hanya baris yang kolom method-nya bernilai "GET". Hasilnya disimpan dalam variabel df_get.

- len(df_get) menunjukkan total baris setelah proses filter GET.

- .head() menampilkan beberapa baris pertama dari data GET.

Hasil Output

- Jumlah data sebelum filter: 1,048,575 baris

- Jumlah data sesudah filter: 1,043,273 baris

Berarti sebagian besar request di dataset ini menggunakan metode GET, yaitu sekitar 99.49% dari keseluruhan data. Ini normal karena GET adalah metode default untuk mengambil resource.

## MENGUBAH REQUEST TIME MENJADI TANGGAL & WAKTU

In [28]:
import pandas as pd

print("Jumlah data sebelum transformasi waktu:", len(df))

# Konversi kolom time menjadi datetime
df['time'] = pd.to_datetime(df['time'], errors='coerce')

# Pisahkan menjadi tanggal dan waktu saja
df['tanggal'] = df['time'].dt.date
df['waktu'] = df['time'].dt.time

print("Jumlah data sesudah transformasi (baris sama):", len(df))

df[['time', 'tanggal', 'waktu']].head(10)

Jumlah data sebelum transformasi waktu: 1048575
Jumlah data sesudah transformasi (baris sama): 1048575


,time,tanggal,waktu
0,1970-01-01 00:00:00.805465029,1970-01-01,00:00:00.805465
1,1970-01-01 00:00:00.805465031,1970-01-01,00:00:00.805465
2,1970-01-01 00:00:00.805465051,1970-01-01,00:00:00.805465
3,1970-01-01 00:00:00.805465053,1970-01-01,00:00:00.805465
4,1970-01-01 00:00:00.805465054,1970-01-01,00:00:00.805465
5,1970-01-01 00:00:00.805465058,1970-01-01,00:00:00.805465
6,1970-01-01 00:00:00.805465068,1970-01-01,00:00:00.805465
7,1970-01-01 00:00:00.805465071,1970-01-01,00:00:00.805465
8,1970-01-01 00:00:00.805465153,1970-01-01,00:00:00.805465
9,1970-01-01 00:00:00.805465157,1970-01-01,00:00:00.805465


## FILTER BERDASARKAN IP ADDRESS & REQUEST PROTOCOL

In [29]:
print("Jumlah data sebelum filter:", len(df))

# ---------------------- FILTER 1: BERDASARKAN IP ADDRESS ----------------------
ip_target = "103.10.20.30"   # ganti IP sesuai yang kamu mau

df_ip = df[df['host'] == ip_target]

print("\n--- FILTER BERDASARKAN IP ADDRESS ---")
print("IP yang difilter :", ip_target)
print("Jumlah data sesudah filter IP :", len(df_ip))
print(df_ip.head())

# ---------------------- FILTER 2: BERDASARKAN REQUEST PROTOCOL ----------------------
# contoh: hanya ambil URL yang menggunakan 'https://'
df_protocol = df[df['url'].str.startswith("https://", na=False)]

print("\n--- FILTER BERDASARKAN REQUEST PROTOCOL (HTTPS) ---")
print("Jumlah data sesudah filter protokol :", len(df_protocol))
print(df_protocol.head())

Jumlah data sebelum filter: 1048575

--- FILTER BERDASARKAN IP ADDRESS ---
IP yang difilter : 103.10.20.30
Jumlah data sesudah filter IP : 0
Empty DataFrame
Columns: [Unnamed: 0, host, time, method, url, response, bytes, tanggal, waktu]
Index: []

--- FILTER BERDASARKAN REQUEST PROTOCOL (HTTPS) ---
Jumlah data sesudah filter protokol : 0
Empty DataFrame
Columns: [Unnamed: 0, host, time, method, url, response, bytes, tanggal, waktu]
Index: []


Pada tahap ini dilakukan dua proses penyaringan data:
1. penyaringan berdasarkan alamat IP tertentu, dan
2. penyaringan berdasarkan protokol request (misalnya HTTPS).

Data awal memiliki 1.048.575 baris, namun setelah difilter berdasarkan IP 103.10.20.30, jumlah baris menjadi 0, karena IP tersebut tidak terdapat pada dataset. Selanjutnya, penyaringan berdasarkan request protocol seperti “https://” juga menghasilkan 0 data, karena dataset NASA 1995 ini hanya menggunakan protokol HTTP, sehingga tidak ada baris yang memenuhi syarat.

Dengan demikian, kedua filter menghasilkan data kosong, namun proses filtering telah berhasil dilakukan sesuai instruksi.